src/Magnum/Math/Geometry/Intersection.h

```diff
@@ -147,14 +149,122 @@ template<class T> bool pointFrustum(const Vector3<T>& point, const Frustum<T>& f
+template<class T> bool pointCone(const Vector3<T>& p, const Vector3<T>& origin, const Vector3<T>& normal, Deg<T> angle);
+
+/**
+@brief Faster version of intersection of a point and a cone
+@param p        The point
+@param origin   Origin of the cone
+@param normal   Normal of the cone
+@param tanAngleSqPlusOne Precomputed portion of the cone intersection equation: `Math::tan(angle/2)+1`
+
+Returns @cpp true @ce if the point is inside the cone.
+
+Uses the result of precomputing @f$x = \tan^2{\theta} + 1@f$.
+*/
+template<class T> bool pointCone(const Vector3<T>& p, const Vector3<T>& origin, const Vector3<T>& normal, T tanAngleSqPlusOne);
+
+/**
+@brief Faster version of intersection of a point and a double cone
+@param p        The point
+@param origin   Origin of the cone
+@param normal   Normal of the cone
+@param tanAngleSqPlusOne Precomputed portion of the cone intersection equation:
+                         @cpp Math::tan(angle/2)+1 @ce
+
+Returns @cpp true @ce if the point is inside the double cone.
+
+Uses the result of precomputing @f$ x = \tan^2{\theta} + 1 @f$.
+*/
+template<class T> bool pointDoubleCone(const Vector3<T>& p, const Vector3<T>& origin, const Vector3<T>& normal, T tanAngleSqPlusOne);
+
+/**
+@brief Intersection of a point and a double cone
+@param p        The point
+@param origin   Origin of the cone
+@param normal   Normal of the cone
+@param angle    Apex angle of the cone
+
+Returns @cpp true @ce if the point is inside the double cone.
+
+@see @ref pointDoubleCone(const Vector3&, const Vector3&, const Vector3&, T)
+*/
+template<class T> bool pointDoubleCone(const Vector3<T>& p, const Vector3<T>& origin, const Vector3<T>& normal, Deg<T> angle);
+
+/**
+@brief Intersection of a sphere and a cone view
+@param sphereCenter Center of the sphere
+@param radius Radius of the sphere
+@param coneView View matrix with translation and rotation of the cone
+@param angle Cone opening angle
+
+Returns @cpp true @ce if the sphere intersects the cone.
+*/
+template<class T> bool sphereConeView(const Vector3<T>& sphereCenter, T radius, const Matrix4<T>& coneView, Rad<T> angle);
+
+/**
+@brief Faster version of intersection of a sphere and a cone view
+@param sphereCenter Center of the sphere
+@param radius Radius of the sphere
+@param coneView View matrix with translation and rotation of the cone
+@param sinAngle Sine of the cone angle
+@param cosAngle Cosine of the cone angle
+@param tanAngle Tangens of the cone angle
+
+Returns @cpp true @ce if the sphere intersects the cone.
+*/
+template<class T> bool sphereConeView(const Vector3<T>& sphereCenter, T radius, const Matrix4<T>& coneView, T sinAngle, T cosAngle, T tanAngle);
+
+/**
+ * @brief Intersection of a sphere and a cone
+ * @param sphereCenter Sphere center
+ * @param radius Sphere radius
+ * @param origin Origin of the cone
+ * @param normal Cone normal
+ * @param angle Cone opening angle (0 < angle < pi).
+ *
+ * Returns `true` if the sphere intersects with the cone.
+ */
```

Comment:
No need to bother with the ` * ` at line starts if the comment block is not indented in any way. Also, `@cpp true @ce` (sorry :D)